keras/backend/theano_backend.py

```diff
@@ -386,7 +386,7 @@ def normalize_batch_in_training(x, gamma, beta,
 def batch_normalization(x, mean, var, beta, gamma, epsilon=0.0001):
     '''Apply batch normalization on x given mean, var, beta and gamma.
     '''
-    if theano.config.device.startswith('cuda') or theano.config.device.startswith('gpu'):
+    if x.ndim < 5 and (theano.config.device.startswith('cuda') or theano.config.device.startswith('gpu')):
```

Comment:
well it could potentially be fewer than 4 dimensions, right? Anyway this just mimics the value check in theano, unfortunately this throws a value error, along with a few other errors, so catching the value error and passing is not an option. I can create an intermediate variable, is that the only issue you see? https://github.com/Theano/Theano/blob/e5c41e766abdbd15c96eab6bd7e2b9136659885c/theano/sandbox/cuda/dnn.py#L2764